src/event/sys/unix/parse.rs

```diff
@@ -527,6 +533,18 @@ fn parse_cb(cb: u8) -> Result<(MouseEventKind, KeyModifiers)> {
     Ok((kind, modifiers))
 }
 
+pub(crate) fn parse_csi_bracketed_paste( buffer: &[u8], ) -> Result<Option<InternalEvent>> {
+    // ESC [ 2 0 0 ~ pasted text ESC 2 0 1 ~
+    assert!(buffer.starts_with(b"\x1B[200~"));
+
+    if !buffer.ends_with(b"\x1b[201~") {
+        Ok(None)
+    } else {
+        let paste = String::from_utf8_lossy(&buffer[6..buffer.len() - 6]).to_string();
```

Comment:
This only works because [the parser is backed by a Vec buffer](https://github.com/crossterm-rs/crossterm/blob/069497b43b6b58b6ff2f3824850b403addf64cae/src/event/source/unix.rs#L181). It'll keep growing till the end of the bracket shows up. Maybe we should put a limit on the max paste? Not sure what that should be though.

Reply:
Lets just allow pasting anything. There is eventually a limit on the clipboard size. Time will tell if we should place a more practical limit

Reply:
Works for me. I was a little worried about memory usage since the source module is being so careful with allocations. I'm happy to not worry about it.